src/socket/unix_socket.cr

```diff
@@ -70,9 +76,12 @@ class UNIXSocket < Socket
   # left.gets # => "message"
   # ```
   def self.pair(type : Type = Type::STREAM) : {UNIXSocket, UNIXSocket}
-    Crystal::System::Socket
-      .socketpair(type, Protocol::IP)
-      .map { |fd| UNIXSocket.new(fd: fd, type: type) }
+    fds, blocking = Crystal::EventLoop.current.socketpair(type, Protocol::IP)
+    fds.map do |fd|
+      sock = UNIXSocket.new(handle: fd, type: type, blocking: blocking)
+      sock.sync = true
+      sock
+    end
```

Comment:
NOTE: we don't currently need the `Crystal::EventLoop#socketpair` to return the blocking arg, as it's only supported on UNIX where we can query the blocking mode, and only win32 needs to save the info as an ivar.